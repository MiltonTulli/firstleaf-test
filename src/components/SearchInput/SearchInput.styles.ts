import { makeStyles, Theme, createStyles } from '@material-ui/core/styles';

export default makeStyles((theme: Theme) =>
  createStyles({
    root: {
      padding: '2px 4px',
      display: 'flex',
      alignItems: 'center',
      width: 300
    },
    input: {
      marginLeft: theme.spacing(1),
      flex: 1
    },
    iconButton: {
      padding: 10
    },
    divider: {
      height: 28,
      margin: 4
    }
  })
);
