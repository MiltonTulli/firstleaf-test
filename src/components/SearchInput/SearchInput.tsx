import React from 'react';
import Paper from '@material-ui/core/Paper';
import InputBase from '@material-ui/core/InputBase';
import Divider from '@material-ui/core/Divider';
import IconButton from '@material-ui/core/IconButton';
import Refresh from '@material-ui/icons/Refresh';
import SearchIcon from '@material-ui/icons/Search';
import CloseIcon from '@material-ui/icons/Close';
import useStyles from './SearchInput.styles';

interface SearchInputProps {
  handleSearch: (value: string) => void;
  filter?: boolean;
}

export function SearchInput({ handleSearch, filter = false }: SearchInputProps) {
  const classes = useStyles();
  const [value, setValue] = React.useState('');
  const handleRefresh = () => {
    setValue('');
    handleSearch('');
  };
  return (
    <Paper component="div" className={classes.root}>
      <InputBase
        data-testid="input"
        className={classes.input}
        placeholder="Search Country"
        inputProps={{ 'aria-label': 'search country' }}
        value={value}
        onChange={e => setValue(e.target.value)}
      />
      {value && (
        <IconButton
          onClick={() => setValue('')}
          className={classes.iconButton}
          aria-label="Clear Text"
        >
          <CloseIcon />
        </IconButton>
      )}
      <Divider className={classes.divider} orientation="vertical" />
      <IconButton
        data-testid="search"
        onClick={() => handleSearch(value)}
        className={classes.iconButton}
        aria-label="search"
      >
        <SearchIcon />
      </IconButton>
      {filter && (
        <IconButton onClick={handleRefresh} className={classes.iconButton} aria-label="refresh">
          <Refresh />
        </IconButton>
      )}
    </Paper>
  );
}
