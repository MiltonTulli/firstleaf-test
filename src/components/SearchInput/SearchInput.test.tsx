import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { SearchInput } from './SearchInput';

describe('<SearchInput>', () => {
  let props;
  beforeEach(() => {
    props = {
      handleSearch: jest.fn(),
      filter: false
    };
  });
  it('Should match snapshot', () => {
    const wrapper = renderer.create(<SearchInput {...props} />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });

  it('Should trigger handleSearch', () => {
    const wrapper = shallow(<SearchInput {...props} />);
    const inputEl = wrapper.find('[data-testid="input"]');
    const submitEl = wrapper.find('[data-testid="search"]');

    const event = { target: { value: 'TEST' } };
    inputEl.simulate('change', event);
    submitEl.simulate('click', {});

    expect(props.handleSearch).toHaveBeenCalled();
  });
});
