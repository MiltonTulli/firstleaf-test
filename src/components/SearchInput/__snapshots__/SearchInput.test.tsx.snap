// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchInput> Should match snapshot 1`] = `
<div
  className="MuiPaper-root makeStyles-root-1 MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    className="MuiInputBase-root makeStyles-input-2"
    data-testid="input"
    onClick={[Function]}
  >
    <input
      aria-label="search country"
      className="MuiInputBase-input"
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="Search Country"
      type="text"
      value=""
    />
  </div>
  
  <hr
    className="MuiDivider-root makeStyles-divider-4 MuiDivider-vertical"
  />
  <button
    aria-label="search"
    className="MuiButtonBase-root MuiIconButton-root makeStyles-iconButton-3"
    data-testid="search"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiIconButton-label"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
        />
      </svg>
    </span>
  </button>
</div>
`;
